src/parser/DASHTree.cpp

```diff
@@ -468,6 +468,40 @@ start(void *data, const char *el, const char **attr)
             }
             dash->currentNode_ |= DASHTree::MPDNODE_SEGMENTTEMPLATE;
           }
+		  else if (strcmp(el, "ContentProtection") == 0) //Inside Representation
```

Comment:
Fair enough.
This might be a bit beyond my skill level, but to me I'm thinking to store the encryption bool inside the representation like so - ```dash->current_adaptationset_->repesentation_->encrypted = true;```

Then move ```dash->adp_pssh_.second = "PROTECTED";``` to ```dash->current_adaptationset_->repesentation_->adp_pssh_.second = "PROTECTED";``` and same for defaultKID_

Then change throughout [here](https://github.com/glennguy/inputstream.adaptive/blob/1c54891491985c0df5e758649324954ea458994f/src/main.cpp#L1101-L1208) to suit. Am I on the right track?